components/UI/FilterMenu/FilterTimeMenu.tsx

```diff
@@ -91,6 +97,19 @@ const FilterTimeMenu: React.FC<FilterTimeMenuProps> = ({ onChangeTime }) => {
     handleClose();
   };
 
+  useEffect(() => {
+    let currentOptionIndex = FilterOptions.findIndex(
+      (option) => option.value === selectedValue
+    );
+
+    if (shouldFetchNextOption && selectedValue !== "all") {
+      setSelectedValue(FilterOptions[currentOptionIndex + 1].value);
```

Comment:
If I selected the last element of the FilterOptions array and no data is returned, an attempt is made to access the non-existent value of the array. In this case, it might make more sense if it went back to the beginning.

Reply:
That's why we're checking for selectedValue !== "all", that means if selected value not equals to last element of option.

Reply:
You can check by this link: https://deprem-yardim-frontend-git-fork-serhanelmali-development-afet.vercel.app/

Reply:
overlooked, looks fine

Reply:
eline saglik